### DISCUSSION TITLE
Comparison to the previous qualification report

### DISCUSSION THREAD

**Discussion by Yuri05** [Open-Systems-Pharmacology/QualificationPlan#165] [Ideas]
When (re-)creating a qualification report - it should be possible to compare the outcome with the previous report version.
Ideally in the fully automated manner.

Let's discuss some ideas how it could be done.

Many qualification plots (GOF, PK-Ratio, DDI-Ratio) produce some quantitative outputs (GMFE, % of points within X-range). Those outputs could be saved in a separate file (e.g. CSV) which can be then easily compared to the previous version (also automatically).

One difficulty here is: if we export such qualification measures for ALL plots and the report structure changes (e.g. some new plots are added, old plots are removed, or just the order of plots changes) - the (automated) comparison will fail.

What we could do: one plot (or several plots) could be marked as "most important" within a qualification plan (e.g. this would be the "main" DDI-/PK-Ratio plot of a report and/or the "main" GOF-plot of a report). All those plots would become a unique plot ID within a qualification plan.

The new "qualification measures" file would contain **only the outputs for those "most important" plots** (e.g. something like below<sup>(*)</sup>) and would be thus invariant in regard to the changes in the qualification plan.

Thoughts?

ID | Title | PK-Parameter | GMFE | % points within 2-fold | % points within 1.5-/"Guest"-Fold
-- | -- | -- | -- | -- | --
1 | DDI Ratio | AUC | 1,2 | 95 | 90
1 | DDI Ratio | CMAX | 1,1 | 100 | 90
2 | GOF Peripheral Venous Blood | --- | 1,4 | 90 | 85
3 | GOF Fraction Excreted to Urine | --- | 1,3 | 99 | 96

<sup>(*)</sup>In this example 2 entries have the same ID because one (DDI-)Plot in a qualification plan might expand into **several** plots in the report.

**Comment by PavelBal** [comment]
Every entry of a report has an ID, right?

Compare elements with the same IDs. If an entry with ID `XY` exists only in one of the reports - just mark them as "added" or "removed".

**Comment by Yuri05** [comment]
>Every entry of a report has an ID, right?

No, there are no such IDs at the moment. They must be defined first.
